### PR TITLE
Fix #3933 Featured dashboard section not in sync with dashboard delete

### LINKED ISSUE #3933: Featured Dashboard list is not updated correctly when delete comes from cards in content tab
### Description
There are different behaviours when deleting a dashboard in the homepage.

- [ ] **Problem 1 - text refers to map while deleting dashboard**
- Find a dashboard card in the Featured section
- Clicking on the trash button
image
- Check the text inside the modal 
image

- [ ] **Problem 2 - deleting dashboard card in one section does not refresh on another section**
- Follow step of problem 1 i.e try to delete dashboard card from featured section
- Click on delete Map button (this is dashboard)
- The deleted card disappear from featured section but remains in the content section
- expect to see content section to update 
- Again go to content section and delete any dashboard card that is also present in featured section
- The card disappear from content section but remains in the featured section
- expect to see the featured/content section update on delete of dashboard card.
- Please note that this behaviour does not exist for map cards, only dashboard card

- [ ] **Problem 3 - "Delete modal" of Featured Section is different then the one in Contents Section**
- Click on trash button from a card in Featured section (save it somewhere like paint or docs)
- Close it via close button
- Click on trash button of same card in of Contents Section
- Compare the two modals

Contents Section
image
vs
Featured Section
image
- [ ] **Problem 4 - Click on transparent area of delete modal originate from content section does nothing**
- Click on trash button of any card in Contents Section
- Clicking in the transparent area of modal that comes from Content Section does not close the modal
- Click on the same transparent area of delete modal comes from featured section do close the modal
- Expect the modal to close on click the transparent area of delete modal

### In case of Bug (otherwise remove this paragraph)

*Browser Affected*
- not browser related

*Steps to reproduce*
see above

*Expected Result*

problem 1) It contains "map" instead of "dashboard"
problem 2) featured list updates by removing the card
problem 3) we can decide to align the two either way with a general message or with some more specific ones in the latter case the text should refer to dashboard instead of map.
problem 4) close the modal

*Current Result*
problem 1) the ui is referring to maps while we are deleting a dashboard
problem 2) Featured/content section update on delete from one section
problem 3) The two modals are different
problem 4) it does nothing

### PR DESCRIPTION
**Pull request by kasongoyo**
## Description
This PR address the following issues;
- Fix dashboard delete confirm modal originate from featured section text content
- Fix sync between featured section and content section of the dashboard when delete dashboard card from either featured or content section
- make delete confirm modal from dashboard section similar to the one of delete map confirm

## Issues
 - #3933 
 - ...

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)

 - [x] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)
see #3933 

**What is the new behavior?**
See description above

**Does this PR introduce a breaking change?** (check one with "x", remove the other)

 - [ ] Yes, and I documented them in migration notes
 - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: web/client/components/resources/modals/ConfirmModal.jsx [hunk 1]
```diff
@@ -17,31 +17,38 @@ module.exports = ({
...
     onConfirm = () => {},
     show,
-    children
-
+    children,
+    className = '',
+    buttonSize,
+    running = false
 }) => {
-    return (<Portal>
```
[the remark this answers is not in the document]
Comment (allyoucanmap) [reply web/client/components/resources/modals/ConfirmModal.jsx]: We have two ConfirmModal in master branch:
- [/components/maps/modals/ConfirmModal.jsx](https://github.com/geosolutions-it/MapStore2/blob/master/web/client/components/maps/modals/ConfirmModal.jsx)
- [/components/resources/modals/ConfirmModal.jsx](https://github.com/geosolutions-it/MapStore2/blob/master/web/client/components/resources/modals/ConfirmModal.jsx)

They have a different structure and we are using the first one for the maps list and the second one for dashboards list.
3rd bullet point requests to use a modal with the same visual ui for both resource types (maps and dashboard).
If we are using the one from maps we should use directly the other component and remove this one instead to create a second copy of the same component.

File: web/client/reducers/dashboards.js [hunk 2]
```diff
@@ -51,6 +52,17 @@ function dashboards(state = {
                 "loading", action.value, state
             ));
         }
+        // dashboard delete use map delete action
+        case MAP_DELETED: {
```
[the remark this answers is not in the document]
Comment (kasongoyo) [reply web/client/reducers/dashboards.js]: The dashboards reducer contain the list of visible dashboards only so it is not possible to have 13 dashboards and only 12 are visible. Besides similar soln has been implemented on maps [see here](https://github.com/geosolutions-it/MapStore2/blob/6fb6df2d07c1e1963d1fbec226201a4617467444/web/client/reducers/maps.js#L165). I would like to assure you that the behaviour you have described the pose the risk is not possible to happen @offtherailz

File: web/client/components/resources/modals/ConfirmModal.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (kasongoyo) [reply web/client/components/resources/modals/ConfirmModal.jsx]: The use of resource Confirm Modal is limited to resource card only i.e it is used in only single place. Well, I agree regarding the reuse of map confirm modal and get rid of resource confirm modal @offtherailz

File: web/client/components/resources/modals/ConfirmModal.jsx [hunk 3]
```diff
@@ -17,31 +17,38 @@ module.exports = ({
...
-            }]}>
-            <div className="ms-alert">
-                <span className="ms-alert-center">
-                    {children}
-                </span>
-            </div>
-        </ResizableModal>
-    </Portal>);
+    const footer = (<span role="footer"><div style={{"float": "left"}}></div>
+    <Button
```
[the remark this answers is not in the document]
Comment (kasongoyo) [reply web/client/components/resources/modals/ConfirmModal.jsx]: Noted

File: web/client/components/resources/modals/ConfirmModal.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (offtherailz) [reply web/client/components/resources/modals/ConfirmModal.jsx]: Ok, didn't get it was resource Specific. @allyoucanmap the idea, not yet compleated, is to use resources components for both maps and dashboards (and other future resources in the future). 

This should be somehow included in #2908 (not explicitly, I'm going to comment it.

File: web/client/reducers/dashboards.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (offtherailz) [reply web/client/reducers/dashboards.js]: Yes, I mean IF on server you have 13, or more dashboards. Then you delete one. You will see only 11 results but you have 12 on the server. See the example: 
Peek 09-08-2019 11-23

This bug affects even maps, as you noticed. There should be a dedicated issue for that. 

In order to not replicate the same problem here, we should simply reload the dashboards with the current search parameters (page index and search text) after a featured resource delete action

I also linked a pull request where I did something similar.

File: web/client/components/resources/modals/ConfirmModal.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (kasongoyo) [reply web/client/components/resources/modals/ConfirmModal.jsx]: So what next here?

File: web/client/reducers/dashboards.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (kasongoyo) [reply web/client/reducers/dashboards.js]: Noted. So I should solve for dashboard and map altogether right? @offtherailz

**Pull request merged**